tests/queries/0_stateless/03202_sz3_codec.sql

```diff
@@ -0,0 +1,216 @@
+-- Tags: no-fasttest
+-- no-fasttest: needs sz3 library
+
+SET allow_experimental_codecs = 1;
+SET cross_to_inner_join_rewrite = 1;
+
+DROP TABLE IF EXISTS tab;
+
+SELECT 'Negative tests';
+
+-- SZ3 can only be created on Float* and Array(Float*) columns
+CREATE TABLE tab (compressed String CODEC(SZ3)) Engine = Memory; -- { serverError BAD_ARGUMENTS }
+CREATE TABLE tab (compressed UInt64 CODEC(SZ3)) Engine = Memory; -- { serverError BAD_ARGUMENTS }
+CREATE TABLE tab (compressed BFloat16 CODEC(SZ3)) Engine = Memory; -- { serverError BAD_ARGUMENTS }
+CREATE TABLE tab (compressed Array(UInt64) CODEC(SZ3)) Engine = Memory; -- { serverError BAD_ARGUMENTS }
```

Comment:
In my testing, `Map(Float64, Float64)` works as well. `Tuple` also works. It would be nice to write very basic tests for them (see my other comment about compact/wide format) but I would also be fine with disallowing these types.